Comp/r-sas_logistic-regr.qmd

```diff
@@ -0,0 +1,373 @@
+|             Convergence criteria             |            NA            |       Default       | Specifies relative difference between deviance \< 1E--8.                                                       |                                                                                                                                                                          |
+|  Confidence interval (CI) estimation method  |         Default          | `confint.default()` | Wald CI                                                                                                        | In `stats::glm` in R, function confint.default() gives the Wald confidence limits; whereas function confint() gives the profile-likelihood limits.                       |
+| Hypothesis tests for regression coefficients |         Default          |       Default       | Wald tests, which are based on estimates for the regression coefficients and its corresponding standard error. |                                                                                                                                                                          |
+
+: Standard Logistic Regression in SAS vs R {#tbl-1}
+
+## Numerical Comparison {#sec-num-comp}
+
+Every effort is made to ensure that the R code employs estimation methods/ optimization algorithms/ other components that closely match (as much as possible) those used in the SAS code. 
+
+### `glm` in R
+
+Note, the default fitting method in `glm` is consistent with the default fitting method in `PROC LOGISTIC` procedure.
+
+-   Default fitting method in `glm` is iteratively reweighted least squares, and the documentation can be found [here](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/glm).
+-   Default fitting method for `PROC LOGISTIC` procedure is Fisher's scoring method, which is reported as part of the SAS default output, and it is equivalent to "Iteratively reweighted least squares" method as reported in this [documentation](https://support.sas.com/documentation/cdl/en/statug/63347/HTML/default/viewer.htm#statug_logistic_sect033.htm).
+
+```{r}
+# stats::glm function
+m1 <- glm(wt_grp ~ age + sex + ph.ecog + meal.cal, data = lung2, family = binomial(link="logit"))
+# model coefficients summary
+summary(m1)$coefficients
+```
+Note, function `confint.default` gives the Wald confidence limits, which is the default option in SAS `PROC LOGISTIC` procedure; whereas `confint` gives the profile-likelihood limits. Conditional odds ratio is calculated by taking the exponential of the model parameters.
+
+```{r}
+cbind(est = coef(m1), 
+          confint.default(m1))
+```
+### `PROC LOGISTIC` in SAS (without firth option)
+
+```{sas, eval = FALSE}
+PROC LOGISTIC DATA=WORK.IMPORT;
+	MODEL WT_GRP(EVENT="weight_gain") = AGE SEX PH_ECOG MEAL_CAL;
+	ods output ESTIMATEs=estimates;
+run;
+```
```

Comment:
Maybe we can refer to the SAS implementation for creation of the WT_GRP variable [here](https://psiaims.github.io/CAMIS/SAS/logistic-regr.html#modelling-the-lung-cancer-data)

Reply:
thanks Orla for the suggestions! I've modified the data loading for SAS refering to lung2 directly as lung2 was created within in qmd. Same approach as in [here](https://psiaims.github.io/CAMIS/SAS/logistic-regr.html#modelling-the-lung-cancer-data)